diff --git a/anykernel.sh b/anykernel.sh
index 400c20b..03fb4b6 100755
--- a/anykernel.sh
+++ b/anykernel.sh
@@ -4,7 +4,7 @@
 ### AnyKernel setup
 # global properties
 properties() { '
-kernel.string=Wild Kernels by TheWildJames aka Morgan Weedman
+kernel.string=nikitos4683 Kernel for OnePlus 13
 do.devicecheck=0
 do.modules=0
 do.systemless=0
@@ -40,8 +40,13 @@ case $kernel_version in
     *) ksu_supported=false ;;
 esac
 
-ui_print " " "  -> Wild Kernels Supported: $ksu_supported"
-$ksu_supported || abort "  -> Non-GKI device, abort."
+ui_print " "
+ui_print "============================================"
+ui_print "      nikitos4683 Kernel Installer"
+ui_print "        OnePlus 13 | OOS16 | GKI"
+ui_print "============================================"
+ui_print "  -> KernelSU-supported base: $ksu_supported"
+$ksu_supported || abort "  -> Unsupported non-GKI base, abort."
 
 # boot install
 split_boot
@@ -53,25 +58,10 @@ else
 fi
 
 ui_print " "
+
-ui_print "WildKernels Telegram Channel:"
-ui_print "https://t.me/WildKernels"
-ui_print " "
-ui_print "WildKernels Website:"
-ui_print "https://wildkernels.dev"
-ui_print " "
-ui_print "Wild_KSU GitHub Repository:"
-ui_print "https://github.com/WildKernels/Wild_KSU"
-ui_print "KernelSU-Next fork focused on customization and root-hiding features!"
+ui_print "Installation complete."
 ui_print " "
-ui_print "GKI_KernelSU_SUSFS GitHub Repository:"
-ui_print "https://github.com/WildKernels/GKI_KernelSU_SUSFS"
-ui_print "GKI kernels with KernelSU and SUSFS."
+ui_print "Kernel repo:"
+ui_print "https://github.com/nikitos4683/OnePlus_KernelSU_SUSFS_OP13"
 ui_print " "
-ui_print "OnePlus_KernelSU_SUSFS GitHub Repository:"
-ui_print "https://github.com/WildKernels/OnePlus_KernelSU_SUSFS"
-ui_print "OnePlus kernels with KernelSU and SUSFS."
-ui_print " "
-ui_print "Samsung_KernelSU_SUSFS GitHub Repository:"
-ui_print "https://github.com/WildKernels/Samsung_KernelSU_SUSFS"
-ui_print "Samsung kernels with KernelSU and SUSFS."
 ui_print " "
