diff --git a/kernel/apk_sign.c b/kernel/apk_sign.c
--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -91,7 +95,7 @@ static bool check_block(struct file *fp, u32 *size4, loff_t *pos, u32 *offset,
 	if (*size4 == expected_size) {
 		*offset += *size4;
 
-#define CERT_MAX_LENGTH 1024
+#define CERT_MAX_LENGTH 2048
 		char cert[CERT_MAX_LENGTH];
 		if (*size4 > CERT_MAX_LENGTH) {
 			pr_info("cert length overlimit\n");
@@ -358,7 +358,10 @@ bool is_manager_apk(char *path)
 	// pkg is `<real package>`
 	if (strncmp(pkg, KSU_MANAGER_PACKAGE, sizeof(KSU_MANAGER_PACKAGE))) {
 		return false;
 	}
 #endif
-	return check_v2_signature(path, EXPECTED_MANAGER_SIZE, EXPECTED_MANAGER_HASH);
+	return (check_v2_signature(path, EXPECTED_MANAGER_SIZE, EXPECTED_MANAGER_HASH)
+			|| check_v2_signature(path, 0x338, "f26471a28031130362bce7eebffb9a0b8afc3095f163ce0c75a309f03b644a1f") // pershoot Manager
+			|| check_v2_signature(path, 0x381, "52d52d8c8bfbe53dc2b6ff1c613184e2c03013e090fe8905d8e3d5dc2658c2e4") // Wild_KSU Manager
+			|| check_v2_signature(path, 0x588, "46ff70534cbd9ff8ae8807bbceda83505cf5ca301dffedcb73722c1b1d6e5b14")); // nikitos4683 Manager
 }
